src/components/modern/LargeTokenValueView.js

```diff
@@ -25,54 +25,71 @@ import { BigNumber } from 'bignumber.js';
 
 // Components
 import { Spacing } from 'components/modern/Layout';
-import Text from 'components/modern/Text';
+import TokenValueView from 'components/modern/TokenValueView';
+import FiatValueView from 'components/modern/FiatValueView';
 
 // Utils
-import { formatTokenValueWithoutSymbol } from 'utils/format';
 import { appFont, spacing } from 'utils/variables';
+import { useThemeColors } from 'utils/themes';
+import { getAssetValueInFiat } from 'utils/rates';
+import { wrapBigNumber } from 'utils/bigNumber';
+
+// Selectors
+import { useFiatCurrency, useChainRates } from 'selectors';
 
 // Types
 import type { TextStyleProp } from 'utils/types/react-native';
-
+import type { Chain } from 'models/Chain';
 
 type Props = {|
   value: ?BigNumber,
+  assetAddress: string,
+  chain: Chain,
   symbol: string,
   style?: TextStyleProp,
 |};
 
 /**
  * Large (& stylized) component to display token value.
  */
-function LargeTokenValueView({ value, symbol, style }: Props) {
-  if (!value) return null;
+function LargeTokenValueView({ value, assetAddress, chain, symbol, style }: Props) {
+  const fiatCurrency = useFiatCurrency();
+  const colors = useThemeColors();
+  const rates = useChainRates(chain);
+  const balanceInFiat = getAssetValueInFiat(value, assetAddress, rates, fiatCurrency);
+  const balanceInFiatNumber = balanceInFiat ? wrapBigNumber(balanceInFiat) : null;
+  if (!value || !balanceInFiatNumber) return null;
 
   return (
     <Container style={style}>
       {/* TokenValue & TokenSymbol are wrapped in plain RN Text in order to make baseline work */}
       <RNText>
```

Comment:
Does this text break reasonably well for long token value?

Reply:
![fiattokenvalue](https://user-images.githubusercontent.com/66996635/128860846-b1be12fe-9b52-4904-ad2a-35b6e96e89ae.png)

Reply:
@divyaBacancy what if we're dealing with 5 digital quantities?

Reply:
@IAmKio 
![iphone8FIiatToken](https://user-images.githubusercontent.com/66996635/129858733-fea1877d-4b13-42e9-8044-faf42f1d9080.png)